n_queen_problem/NQueenProblem.java

```diff
@@ -0,0 +1,53 @@
+import java.util.Arrays;
+
+public class NQueenProblem { 
+    private static String placeQueenOnBoard(int Qi, int[] board) {
+        //base case
+        if (Qi == board.length) {
+            return Arrays.toString(board);
+        } 
+        else if (board.length == 2 || board.length == 3) {
+            return "No possible board.";
+        }
+        else {
+        //try to put the ith Queen (Qi) in all of the columns
+            for (int column = 0; column < board.length; column++) {
+                if (isSafePlace(column, Qi, board)) {
+                    board[Qi] = column;
+                    //then place remaining queens.
+                    placeQueenOnBoard(Qi + 1, board);
```

Comment:
This call here returns a String which contains our answer hence needs to be appended to a single variable containing all the answers.